File: 프로그래머스/lv1/86491. 최소직사각형/최소직사각형.js
function solution(sizes) {
  let maxVal0 = 0;
  let maxVal1 = 0;
  for (let i = 0; i < sizes.length; i++) {
    const [a, b] = sizes[i].sort((a, b) => b - a);
    maxVal0 = Math.max(maxVal0, a);
    maxVal1 = Math.max(maxVal1, b);
  }

  return maxVal0 * maxVal1;
}

File: 프로그래머스/1/86491. 최소직사각형/최소직사각형.py
def solution(sizes):
    w = 0
    h = 0
    for size in sizes:
        if size[0] > size[1]:
            size[0], size[1] = size[1], size[0]
        if size[0] > w:
            w = size[0]
        if size[1] > h:
            h = size[1]
    return w * h